plugins/gatsby-plugin-auto-index-pages/gatsby-node.js

```diff
@@ -146,10 +178,49 @@ exports.createPages = async ({ actions, graphql, reporter }, pluginOptions) => {
         });
       });
     }
+  });
 
-    locales.forEach(({ localizedPath }) => {
+  locales.forEach(({ localizedPath }) => {
+    visit(list, 'directory', (dir) => {
+      const slug = `/${dir.path}`;
       const localizedSlug = path.join(`/${localizedPath}`, slug);
 
+      if (skippedDirectories.includes(dir.path)) {
+        return [visit.SKIP];
+      }
+
+      if (existingPaths.includes(slug)) {
+        return;
+      }
+
+      dir.children
+        .filter((child) => child.type === 'directory')
+        .forEach((subDir) => {
+          if (subDir.children) {
```

Comment:
We might be able to get rid of this `if` statement to help with some of the indentation:

```js
dir.children
  .filter((child) => child.type === 'directory)
  .filter((subDir) => subDir.children && subDir.children.length)
  .forEach(({ children }) => {
    // do your filtering and whatnot
  });
```

You might be able to reduce nesting even further to remove that nested loop:

```js
dir.children
  .filter((child) => child.type === 'directory)
  .filter((subDir) => subDir.children && subDir.children.length)
  .flatMap(({children}) => children)
  .filter((child) => child.type === 'file')
  .forEach((child) => {
    // do your child-specific stuff here
  });
```

Reply:
And if you wanna go _super_ functional:

```js
const isDirectory = ({ type }) => type === 'directory';
const isFile = ({ type }) => type === 'file';
const hasChildren = (dir) => dir.children && dir.children.length;

dir.children
  .filter(isDirectory)
  .filter(hasChildren)
  .flatMap(prop('children'))
  .filter(isFile)
  .forEach((child) => {
    // do your child-specific stuff here
  });

Reply:
much neater! just pushed that up